src/api/app/models/concerns/staging_project.rb

```diff
@@ -52,17 +52,19 @@ def update_self_referencing_repositories!(old_project)
   end
 
   def classified_requests
-    requests = (requests_to_review | staged_requests.includes(:reviews)).map do |request|
+    my_staged_requests = staged_requests.includes(:bs_request_actions).to_a
+    ActiveRecord::Associations::Preloader.new.preload(my_staged_requests, :reviews, Review.where.not(state: :accepted,
```

Comment:
I would take a look on this issue @krauselukas  https://github.com/rails/rails/issues/32140. I'm not sure if the `ActiveRecord::Associations::Preloader` is already a public API..

Reply:
@vpereira hmmm, wasn't aware of it thanks, I will try to find another way. Very unfortunate, the Preloader comes in handy :)

Reply:
@krauselukas I didn't know it until now, thank you :)